Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -192,6 +193,24 @@ static inline uint64_t argumentClampedIndexFromStartOrEnd(JSGlobalObject* global
     return indexDouble > length ? length : static_cast<uint64_t>(indexDouble);
 }
 
+static inline int64_t argumentUnclampedIndexFromStartOrEnd(JSGlobalObject* globalObject, JSValue value, uint64_t length, uint64_t undefinedValue = 0)
+{
+    if (value.isUndefined())
+        return undefinedValue;
+
+    if (LIKELY(value.isInt32())) {
+        int64_t indexInt64 = value.asInt32();
+        if (indexInt64 < 0)
+            indexInt64 += length;
+        return indexInt64;
+    }
+
+    double indexDouble = value.toIntegerOrInfinity(globalObject);
+    if (indexDouble < 0)
+        indexDouble += length;
+    return static_cast<int64_t>(indexDouble);
```

Comment:
Casting infinity to `int64_t` is undefined behavior.

Reply:
I've modified it so that if `indexDouble` is `Infinity`, it's approximated by the maximum or minimum `int64_t` value. What do you think?

Reply:
That sounds fine to me!